### PR TITLE
End of support for PHP 7.4

### LINKED ISSUE #235: PHP version mis-match -  Packagist  vs. Github Actions
## Description

Currently, Github Actions file is building 8.1 & 8.2, whereas packagist repo allows 7.4 and 8.0.   This leads to confusion with enterprise CI lifecycle -   eg. Artifactory reverses-proxies a known source packagist. 

## Steps to Reproduce
See version on packagist - 

> https://packagist.org/packages/apigee/apigee-client-php
> requires php: ^7.4 || ^8.0

See github actions - https://github.com/apigee/apigee-client-php/blob/2.x/.github/workflows/php-test-workflow.yml#L22
>                -   php: '8.1'
>                    dependencies-preference: " "

## Actual Behavior
N/A  - Lifecycle/Release Management

## Expected Behavior
Keep github-actions & composer.json / packagist in-sync -   Example,  Looks like php 8.0 and php 8.1 is current supported version?  If so, please drop 7.4 and add 8.1

## Screenshots
https://github.com/apigee/apigee-client-php/blob/2.x/composer.json
vs. 
https://packagist.org/packages/apigee/apigee-client-php

## Notes
N/A

## Version Info
N/A

### PR DESCRIPTION
**Pull request by shishir-intelli**
Closes #235
End of support for PHP 7.4

### CODE REVIEW

**Comment by shishir-intelli** [comment]
Rolled back PHP 8.2  GitHub action test due to  league/period library compatibility issue with PHP 8.2
Created separate issue here https://github.com/apigee/apigee-client-php/issues/240

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "homepage": "http://github.com/apigee/apigee-client-php",
     "license": "Apache-2.0",
     "require": {
-        "php": "^7.4 || ^8.0",
+        "php": "^8.0",
```
Comment (mxr576) [inline composer.json]: Be more specific about actually supported (and tested) PHP versions.

See: https://github.com/apigee/apigee-edge-drupal/pull/816


```suggestion
        "php": "~8.0.0 || ~8.1.0",
```

**Review by giteshk — CHANGES_REQUESTED** [review]
Please hold off on this change.

**Review by giteshk — APPROVED** [review]

**Comment by giteshk** [comment]
We are ready to end support for 7.4 with the latest changes for AppGroup needing PHP 8.0 minimum

**Pull request merged**